crates/core/tedge/src/cli/certificate/renew.rs

```diff
@@ -0,0 +1,78 @@
+use super::error::CertError;
+use crate::command::Command;
+use camino::Utf8PathBuf;
+use certificate::KeyCertPair;
+use certificate::NewCertificateConfig;
+use certificate::PemCertificate;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::prelude::*;
+use std::path::Path;
+use tedge_utils::paths::set_permission;
+use tedge_utils::paths::validate_parent_dir_exists;
+
+pub struct RenewCertCmd {
+    /// The path where the device certificate will be stored
+    pub cert_path: Utf8PathBuf,
+}
```

Comment:
First hint that something is wrong: the command is given no path to the current private key and therefore has no chance to create a self-signed certificate for the *same* private key.